parastell/magnet_coils.py

```diff
@@ -401,164 +386,110 @@ class MagnetCoil(object):
-                align its normal with the initial point tangent (deg).
-        """
-        # Determine orientation of thickness vector after cross-section was
-        # oriented along filament origin tangent
-
-        # Compute part of thickness vector parallel to rotation axis
-        t_vec_par = normalize(np.inner(t_vec, rot_axis) * rot_axis)
-        # Compute part of thickness vector orthogonal to rotation axis
-        t_vec_perp = normalize(t_vec - t_vec_par)
-
-        # Compute vector othogonal to both rotation axis and orthogonal
-        # part of thickness vector
-        orth = normalize(np.cross(rot_axis, t_vec_perp))
-
-        # Determine part of rotated vector parallel to original
-        rot_par = np.cos(rot_ang_norm)
-        # Determine part of rotated vector orthogonal to original
-        rot_perp = np.sin(rot_ang_norm)
+    def __init__(self, coords, tangents, center_of_mass, width, thickness):
 
-        # Compute orthogonal part of thickness vector after rotation
-        t_vec_perp_rot = rot_par * t_vec_perp + rot_perp * orth
-        # Compute thickness vector after rotation
-        t_vec_rot = normalize(t_vec_perp_rot + t_vec_par)
-
-        # Orient cross-section in its plane such that it faces the global origin
-
-        # Extract initial path point
-        pos = cubit.vertex(path_origin).coordinates()
-
-        # Project position vector onto cross-section
-        pos_proj = normalize(pos - np.inner(pos, norm) * norm)
-
-        # Compute angle by which to rotate cross-section such that it faces the
-        # origin
-        rot_ang_orig = np.arccos(np.inner(pos_proj, t_vec_rot))
-
-        # Re-orient rotated cross-section such that thickness vector faces
-        # origin
-        cubit.cmd(
-            f"rotate Surface {surf_id} angle {np.rad2deg(rot_ang_orig)} about "
-            "origin 0 0 0 direction " + " ".join(str(i) for i in norm)
-        )
+        self.coords = coords
+        self.tangents = tangents
+        self.center_of_mass = center_of_mass
+        self.width = width
+        self.thickness = thickness
 
     def create_magnet(self):
-        """Creates magnet coil volumes in cubit.
+        """Creates a single magnet coil CAD solid in CadQuery.
 
         Returns:
-            volume_id (int): magnet volume ids in cubit
+            coil (object): cq.Solid object representing a single magnet coil.
         """
-        # Cross-section inititally populated with thickness vector
-        # oriented along x axis
-        t_vec = np.array([1, 0, 0])
+        tangent_vectors = [
+            cq.Vector(tuple(tangent)) for tangent in self.tangents
+        ]
 
-        # Create cross-section for sweep
-        cubit.cmd(f"create surface " + self.shape_str + " zplane")
+        # Define coil filament path normals such that they face the filament
+        # center of mass
+        normal_dirs = np.array([i - self.center_of_mass for i in self.coords])
```

Comment:
If this is a numpy array, then this line is
```suggestion
        normal_dirs = self.coords - self.center_of_mass
```

Reply:
They are numpy arrays, I just wasn't careful when rewriting from an earlier version.